Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/FilterScreen.kt

```diff
@@ -0,0 +1,261 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.example.jetsnack.R
+import com.example.jetsnack.model.SnackRepo
+import com.example.jetsnack.ui.components.FilterChip
+import com.example.jetsnack.ui.theme.JetsnackTheme
+import com.google.accompanist.flowlayout.FlowRow
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun FilterScreen(
+    onDismiss: () -> Unit
+) {
+    var sortChanged by remember { mutableStateOf(false) }
+    Dialog(onDismissRequest = onDismiss) {
+
+        val priceFilters = remember { SnackRepo.getPriceFilters() }
+        val categoryFilters = remember { SnackRepo.getCategoryFilters() }
+        val lifeStyleFilters = remember { SnackRepo.getFilters() }
+        Scaffold(
+            topBar = {
+                TopAppBar(
+                    navigationIcon = {
+                        IconButton(onClick = onDismiss) {
+                            Icon(
+                                imageVector = Icons.Filled.Close,
+                                contentDescription = null
+                            )
+                        }
+                    },
+                    title = {
+                        Text(
+                            text = stringResource(id = R.string.label_filters),
+                            modifier = Modifier.fillMaxWidth(),
+                            textAlign = TextAlign.Center,
+                            style = MaterialTheme.typography.h6
+                        )
+                    },
+                    actions = {
+                        var resetEnabled = sortChanged
+                        IconButton(
+                            onClick = { /* TODO: Open search */ },
+                            enabled = resetEnabled
+                        ) {
+                            val alpha = if (resetEnabled) 1f else 0.5f
+
+                            Text(
+                                text = stringResource(id = R.string.reset),
+                                style = MaterialTheme.typography.body2,
+                                color = JetsnackTheme.colors.textLink,
+                                modifier = Modifier.alpha(alpha)
+                            )
+                        }
+                    },
+                    backgroundColor = JetsnackTheme.colors.uiBackground
+                )
+            },
+            backgroundColor = JetsnackTheme.colors.uiBackground,
+            modifier = Modifier.padding(0.dp)
+        ) {
+            Column(
```

Comment:
No scrollable container—how does work on shorter devices/landscape?

Reply:
Just added the vertical scroll to the column